### PR TITLE
Fix various warnings

### LINKED ISSUE #397: Fix thrown warnings in dimension, elements, diversity
### Describe the problem

1. [`momepy/dimension.py:464: FutureWarning: Unlike other reduction functions (e.g. skew, kurtosis)...`](https://github.com/pysal/momepy/actions/runs/3039305780/jobs/4894087800#step:6:126)
   * update [`scipy.stats.mode`](https://github.com/pysal/momepy/blob/main/momepy/dimension.py#L464) call
2. [`momepy/elements.py:617: UserWarning: CRS mismatch between the CRS...`](https://github.com/pysal/momepy/actions/runs/3039305780/jobs/4894087800#step:6:152)
   * [This here](https://github.com/pysal/momepy/blob/main/tests/test_elements.py#L120-L124) seems remove the CRS attribute of `streets`, so it should be set again. Might this be a deeper issue with `pandas`/`geopandas`?
3. [`momepy/diversity.py:125: DeprecationWarning: invalid escape sequence '\s'`](https://github.com/pysal/momepy/actions/runs/3039305780/jobs/4894087800#step:6:122)
   * correct troublemaker escape sequences in `diversity.Theil` docstring.

### Steps to reproduce

For (1):
```python
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:43:44) [Clang 13.0.1 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas, scipy

In [2]: vals = pandas.Series([0, 0, 1])

In [3]: scipy.stats.mode(vals)[0][0]
<ipython-input-3-54172f5dfb41>:1: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  scipy.stats.mode(vals)[0][0]
Out[3]: 0

In [4]: scipy.stats.mode(vals, keepdims=True)[0][0]
Out[4]: 0

In [5]: scipy.stats.mode(vals, keepdims=False)[0]
Out[5]: 0

```

### Versions of your packages

```momepy='0.5.3+18.g6f9a2cf.dirty'```

### Your operating system

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jGaboardi**
Closes #397.

### CODE REVIEW

File: tests/test_elements.py [hunk 1]
```diff
@@ -123,6 +123,7 @@ def test_Blocks_inner(self):
             .buffer(20)
             .exterior
         )
+        streets = streets.set_crs(self.df_tessellation.crs)
```
Comment (martinfleis) [inline tests/test_elements.py:126]: The CRS loss is caused by an incorrect indexing we do. The line 120 should be `streets.loc[35, 'geometry'] = (`, not `streets.loc[35] = (`.

Comment (jGaboardi) [reply tests/test_elements.py:126]: And the incorrect indexing we do is not on purpose for the test, correct?

Comment (martinfleis) [reply tests/test_elements.py:126]: No, we are just adding an line geometry fully enclosed within another one to check that the inner rings are correctly resolved.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Pull request merged**